src/simulators/statevector/chunk/chunk.hpp

```diff
@@ -0,0 +1,256 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2018, 2019, 2020.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+
+#ifndef _qv_chunk_hpp_
+#define _qv_chunk_hpp_
+
+#include "simulators/statevector/chunk/device_chunk_container.hpp"
+#include "simulators/statevector/chunk/host_chunk_container.hpp"
+
+namespace AER {
+namespace QV {
+
+
+
+
+//============================================================================
+// chunk class
+//============================================================================
+template <typename data_t>
+class Chunk 
+{
+protected:
+  mutable ChunkContainer<data_t>* chunk_container_;   //pointer to chunk container
+  Chunk<data_t>* cache_;                //pointer to cache chunk on device
```

Comment:
If possible is preferable to use smart pointers. For `chunk` and `chunk_container` I have been able to encapsulate them in `shared_ptr` (https://github.com/doichanj/qiskit-aer/pull/3)

Reply:
done